File: src/main/java/no/nav/tjenestepensjon/simulering/service/StillingsprosentService.java
package no.nav.tjenestepensjon.simulering.service;

import java.util.List;
import java.util.Map;

import no.nav.tjenestepensjon.simulering.domain.Stillingsprosent;
import no.nav.tjenestepensjon.simulering.domain.TPOrdning;
import no.nav.tjenestepensjon.simulering.domain.TpLeverandor;
import no.nav.tjenestepensjon.simulering.exceptions.DuplicateStillingsprosentEndDateException;
import no.nav.tjenestepensjon.simulering.exceptions.MissingStillingsprosentException;

public interface StillingsprosentService {

    StillingsprosentResponse getStillingsprosentListe(String fnr, Map<TPOrdning, TpLeverandor> tpOrdningAndLeverandorMap);
    TPOrdning getLatestFromStillingsprosent(Map<TPOrdning, List<Stillingsprosent>> map) throws DuplicateStillingsprosentEndDateException, MissingStillingsprosentException;
}

File: src/main/kotlin/no/nav/tjenestepensjon/simulering/service/StillingsprosentService.kt
package no.nav.tjenestepensjon.simulering.service

import no.nav.tjenestepensjon.simulering.exceptions.DuplicateStillingsprosentEndDateException
import no.nav.tjenestepensjon.simulering.exceptions.MissingStillingsprosentException
import no.nav.tjenestepensjon.simulering.model.v1.domain.FNR
import no.nav.tjenestepensjon.simulering.model.v1.domain.TPOrdning
import no.nav.tjenestepensjon.simulering.TPOrdningStillingsprosentMap
import no.nav.tjenestepensjon.simulering.TPOrdningTpLeverandorMap

interface StillingsprosentService {
    fun getStillingsprosentListe(fnr: FNR, tpOrdningAndLeverandorMap: TPOrdningTpLeverandorMap): StillingsprosentResponse
    @Throws(DuplicateStillingsprosentEndDateException::class, MissingStillingsprosentException::class)
    fun getLatestFromStillingsprosent(map: TPOrdningStillingsprosentMap): TPOrdning
}